### ISSUE TITLE
Is yakGPT a little more expensive in some way?

### ISSUE THREAD

**Issue by huangpufan** [yakGPT/yakGPT#72]
When I compared different projects on github about apikey using openai, I found that they are not the same in terms of GPT-4 usage costs.

It seems that yakGPT costs a little bit more, basically around 0.09$ for one issue, and is relatively stable. A simpler program fluctuates between 0.04$ and 0.09$.

I'm curious about this: their questions and answers are close in terms of word count, so what accounts for the different costs of utilizing apikey for different open source projects?

**Comment by dev-msp** [comment]
I actually think this is a bug and it's not calculating GPT-4 price correctly.

GPT-4 pricing is unique compared to the other OpenAI models. It's $0.03 / 1k for prompt tokens (which includes all previous messages) and $0.06 / 1k for generated tokens. But YakGPT just uses a flat rate of $0.06.

The effect of this is that the price of a chat will appear to increase more quickly as messages accumulate.

Fixing this would require a refactor of the chat API logic, but it probably wouldn't be too complicated.

**Comment by dev-msp** [comment]
Added a PR for this, #73.

**Comment by huangpufan** [comment]
Thank you for your prompt response.

I was not clear and comprehensive enough in describing the questions before, in fact there were two questions before.

Question 1:
In the website, the billing about GPT4 shows several times more than the actual cost. I found this out by looking at the backend page of openai and comparing the actual cost of consumption. Today I did another experiment, here is a question I asked via yakgpt, the overhead shown by yakgpt and the overhead shown by the official website.

openai:
![image](https://user-images.githubusercontent.com/15962909/235827706-c785aca0-ac3c-4429-90a7-c7f4fd8502e3.png)
yakGPT:
![image](https://user-images.githubusercontent.com/15962909/235828705-b46ba705-616a-444a-a0de-9051ee207676.png)

Question 2:
On the actual cost comparison I observed from the openai backend, the cost of yakgpt is slightly higher than [this project](https://github.com/anse-app/chatgpt-demo). This is the question I wanted to ask yesterday.

But in fact I didn't repeat this question a lot to make sure the problem exists, I just tried three to five questions, or maybe the problem is no longer there in mass usage.
